From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pancakse <pancakseyt@gmail.com>
Date: Mon, 6 Nov 2023 08:57:44 -0500
Subject: [PATCH] Deepslate config files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 7eae8683c0d540ba049dddaed8737e051d6f60db..193aa8c3e206e05e3dd1358ad57281af4031aa4d 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2202,6 +2202,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Deepslate start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getDeepslateConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Deepslate end
+
         /**
          * Sends the component to the player
          *
