From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pancakse <pancakseyt@g.com>
Date: Tue, 21 Nov 2023 21:59:57 +0700
Subject: [PATCH] Do not process chat/commands before player has joined


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 552cc2ee9b77958216a237a92e25317551649b56..96b5a5378b65d1f8efb3a7a5bc04ea688f69b6f8 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -6,6 +6,7 @@ import com.mojang.authlib.GameProfile;
 import com.mojang.brigadier.ParseResults;
 import com.mojang.brigadier.StringReader;
 import com.mojang.logging.LogUtils;
+import org.deepslatemc.deepslate.configuration.DeepslateConfig;
 import it.unimi.dsi.fastutil.ints.Int2ObjectMap.Entry;
 import it.unimi.dsi.fastutil.ints.Int2ObjectMaps;
 import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
@@ -2316,6 +2317,8 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
             this.disconnect(Component.translatable("multiplayer.disconnect.out_of_order_chat"), org.bukkit.event.player.PlayerKickEvent.Cause.OUT_OF_ORDER_CHAT); // Paper - kick event causes
             }); // Paper - push to main
             return Optional.empty();
+        } else if (player.joining && DeepslateConfig.doNotProcessChatCommands) { // Deepslate - EMC - do not handle chat messages before they joined
+            return Optional.empty();
         } else {
             Optional<LastSeenMessages> optional = this.unpackAndApplyLastSeen(acknowledgment);
 
diff --git a/src/main/java/org/deepslatemc/deepslate/configuration/DeepslateConfig.java b/src/main/java/org/deepslatemc/deepslate/configuration/DeepslateConfig.java
index 4d30c26a1803eb86f9c98c1cff616d16458efa6d..dcd987d30f8bb939f7ea7cfbccfafd486b77d7b6 100644
--- a/src/main/java/org/deepslatemc/deepslate/configuration/DeepslateConfig.java
+++ b/src/main/java/org/deepslatemc/deepslate/configuration/DeepslateConfig.java
@@ -157,4 +157,9 @@ public class DeepslateConfig {
     private static void allowAnyUsername() {
         allowAnyUsername = getBoolean("settings.player.allow-any-username", allowAnyUsername);
     }
+
+    public static boolean doNotProcessChatCommands = true;
+    private static void  doNotProcessChatCommands() {
+        doNotProcessChatCommands = getBoolean("settings.do-not-process-chat-commands", doNotProcessChatCommands);
+    }
 }
\ No newline at end of file
