From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pancakse <pancakseyt@gmail.com>
Date: Wed, 22 Nov 2023 18:35:02 +0700
Subject: [PATCH] Use git hash instead build number


diff --git a/build.gradle.kts b/build.gradle.kts
index 592bc0467d356a6a04cf7df4b1da3f40a6cd435f..5b8200610c9ab7338664dc076245aa22e29d685a 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -82,7 +82,7 @@ tasks.jar {
     manifest {
         val git = Git(rootProject.layout.projectDirectory.path)
         val BUILD_NUMBER = git("rev-parse", "--short=7", "HEAD").getText().trim()
-        val implementationVersion = System.getenv("$gitHash") ?: "\"$BUILD_NUMBER\""
+        val implementationVersion = /*System.getenv("$gitHash") ?:*/ "\"BUILD_NUMBER\"" // Deepslate
         val date = git("show", "-s", "--format=%ci", BUILD_NUMBER).getText().trim() // Paper
         val gitBranch = git("rev-parse", "--abbrev-ref", "HEAD").getText().trim() // Paper
         attributes(
